analyzers/src/SonarAnalyzer.CSharp/Rules/SetLocaleForDataTypes.cs

```diff
@@ -113,20 +112,20 @@ public sealed class SetLocaleForDataTypes : SonarDiagnosticAnalyzer
                 ? semanticModel.GetSymbolInfo(objectCreation.Type).Symbol
                 : semanticModel.GetSymbolInfo(constructorCall).Symbol?.ContainingType;
 
-        private static SyntaxNode GetAssignmentTargetVariable(SyntaxNode objectCreation) =>
+        private static ImmutableArray<SyntaxNode> GetAssignmentTargetVariables(SyntaxNode objectCreation) =>
             objectCreation.GetFirstNonParenthesizedParent() switch
             {
-                AssignmentExpressionSyntax assignment => assignment.Left,
-                EqualsValueClauseSyntax equalsClause when equalsClause.Parent.Parent is VariableDeclarationSyntax variableDeclaration => variableDeclaration.Variables.Last(),
-                _ => null
+                AssignmentExpressionSyntax assignment => assignment.AssignmentTargets(),
+                EqualsValueClauseSyntax equalsClause when equalsClause.Parent.Parent is VariableDeclarationSyntax variableDeclaration =>
+                    variableDeclaration.Variables.Select(x => (SyntaxNode)x).ToImmutableArray(),
```

Comment:
I was wrong here: `variableDeclaration.Variables.Last()` is actually right. The "=" only assigns to the last variable. See e.g. https://sharplab.io/#v2:EYLgtghglgdgNAFxAJwK7wCYgNQB8ACATAIwCwAUPgMwAERNAwjQN4U3t2GEtsd8D0/WAhpQaAXhrE4NAFYSahANy8+7YaJnzJy1WvzEAnAAoAJACIo45lAC+W67NvmAlCvJ9bFW0A==